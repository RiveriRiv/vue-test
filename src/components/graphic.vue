<template>
	<table><tr v-for="n in temperatures" v-bind:key="n">{{n}}<td v-bind:class="{colored: (n <= temperature && y === day)}" v-for="y in daysNumbers" v-bind:key="y"></td></tr></tr><tr><td></td><td v-for="day in days" v-bind:key="day">{{day}}</td></tr></table>
</template>

<script>
export default {
	data: function() {
    return {
      temperatures: [40, 35, 30, 25, 20, 15, 10, 5, 0],
      daysNumbers: [0, 1, 2, 3, 4, 5, 6],
      days: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
    }
  },
  computed: {
        town() {
            return this.$store.state.town
        },
        temperature() {
            return this.$store.state.temperature
        },
        day() {
            return this.$store.state.day
        }
    }
	}
</script>