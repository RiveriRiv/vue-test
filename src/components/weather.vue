<template>
	<table id="table"><tr><th class="th">Town</th><th class="th">Temperature</th></tr><tr><td class="td">{{town}}</td><td class="td">{{temperature}}</td></tr></table>
</template>
<script>
export default {
	computed: {
        town() {
            return this.$store.state.town
        },
        temperature() {
            return this.$store.state.temperature
        }
    }
	}
</script>