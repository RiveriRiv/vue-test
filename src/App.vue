<template>
  <div id="app">
  <label for="towns">Choose town:</label>

<select @change="getWeather" v-model="town" name="towns" id="towns">
  <option selected value="Moscow">Moscow</option>
  <option value="Saint-Petersburg">Saint-Petersburg</option>
</select>
  <weather></weather>
<p>{{town}}</p>
<graphic></graphic>
</div>
</template>

<style lang="scss">
#table, .th, .td {
  border: 2px solid black;
}

table {
  border-collapse: collapse;
}

.colored {
  background-color: red;
  width: 20px;
  line-height: 20px;
}
</style>
<script>
import weather from "@/components/weather.vue";
import graphic from "@/components/graphic.vue";


export default {
  name: "components",
  components: {
    weather,
	graphic
  },
   methods: {
        getWeather() {
            this.$store.dispatch('updateTown', this.town);
            this.$store.dispatch('updateTemperature', this.town);
        }
    }
};
</script>
