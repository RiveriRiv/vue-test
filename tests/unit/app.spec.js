import { createLocalVue } from '@vue/test-utils'
import { mount } from "@vue/test-utils";
import App from "@/App.vue";
import Vuex from 'vuex';
import graphic from "@/components/graphic.vue";
import weather from "@/components/weather.vue";

const localVue = createLocalVue()

localVue.use(Vuex)

describe("App.vue", () => {
  let actions
  let store
  let state

  beforeEach(() => {
	state = {
      town: 'Moscow',
	  temperature: 20,
	  day: 2
    }

    actions = {
      updateTown: jest.fn(),
      updateTemperature: jest.fn()
    }
    store = new Vuex.Store({
	  state,
      actions
    })
  })

  it("is vue instance", () => {
    const wrapper = mount(App, {store, localVue});
    expect(wrapper.isVueInstance()).toBe(true)
  });

  it("matches snapshot", () => {
    const wrapper = mount(App, {store, localVue});
     expect(wrapper.html).toMatchSnapshot();
  });

  it("contains other components", () => {
    const wrapper = mount(App, {store, localVue});
    expect(wrapper.contains(graphic)).toBe(true);
	expect(wrapper.contains(weather)).toBe(true);
  });
});