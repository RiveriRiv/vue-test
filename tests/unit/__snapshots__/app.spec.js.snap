// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue matches snapshot 1`] = `[Function]`;

exports[`App.vue renders props.msg when passed 1`] = `[Function]`;
